docs/security-policy.md

```diff
@@ -0,0 +1,112 @@
+NSX-T Distributed Firewall rules, then updates CR status with realized state.
+
+## CRD Design
+
+SecurityPolicy is Namespaced scope CRD, admins could apply CR within a namespace to
+configure security of workloads which are attached to NSX-T networking.
+Compared with standard K8s NetworkPolicies, SecurityPolicy CRD adds `VMSelector` to
+select VMs, it also introduces some rule syntaxes including Drop and Reject actions,
+priority, rule level selectors(`appliedTo`). 
+
+An example of SecurityPolicy CR:
+
+```yaml
+apiVersion: nsx.vmware.com/v1alpha1
+kind: SecurityPolicy
+metadata:
+  name: db-isolation
+  namespace: prod-ns
+spec:
+  priority: 1
+  appliedTo:
+    - vmSelector:
+        matchLabels:
+          role: db
+  rules:
+    - direction: in
+      action: allow
+      sources:
+        - namespaceSelector:
+            matchLabels:
+              role: control
+        - podSelector:
+            matchLabels:
+              role: frontend
+      ports:
+        - protocol: TCP
+          port: 8000
+    - direction: out
+      action: allow
+      destinations:
+        - podSelector:
+            matchLabels:
+              role: dns
+      ports:
+        - protocol: UDP
+          port: 53
+    - direction: in
+      action: drop
+    - direction: out
+      action: drop
+status:
+  conditions:
+    - type: "Ready"
+      status: True
+      reason: "SuccessfulRealized"
+```
+
+The example CR defines security rules for VMs which match the label "role: db"
+in Namespace prod-ns. The first rule allows traffic from Namespaces match label
+"role: control" or Pods match label "role: frontend" in Namespace prod-ns to
+access through TCP with port 8000. The second rule allows the selected VMs to
+access Pods with label "role: dns" through UDP with port 53. The third and forth
+rules are to drop any other ingress and egress traffic to/from the selected VMs.
+
+Below are explainations for the fields:
+
+**spec**: defines all the configurations for a SecurityPolicy CR.
+
+**priority**: defines the order of policy enforcement within a cluster, the range
+is from 0 to 1000. Lower value has higher priority between policies.
+
+**appliedTo**: is a list of policy targets to apply rules. As the CRD is namespaced
+scope, `vmSelector` or `podSelector` will be selected from the Namespace where the
+CR is created. `vmSelector` and `podSelector` cannot be in one entry as it would
+not select any workload. We can also have `appliedTo` in each rule entry, but if
+there is policy level `appliedTo`, it will take precedence over rule level.
+
+**rules**: is a list of policy rules. The relative priority is based on the rule
+order in the list, rules in the front have higher priority than rules in the end.
+
+**action**: specifies the action to be applied on the rule, including 'Allow',
+'Drop' and 'Reject'.
+
+**Direction**: is the direction of the rule, including 'In' or 'Ingress', 'Out'
```

Comment:
It's better that all the `**title**` have the same style? e.g. spec/priority/.../direction/... or Spec/Priority/.../Direction/...